### PR TITLE
Fix duplicated rich text input inside choosers

### LINKED ISSUE #12002: RichTextField duplicates in Document Chooser upload tab
### Issue Summary

When using RichTextField in custom document model, each opening-closing of Document Chooser dialog leads to duplication of RichTextField in upload tab.

Screenshot

### Steps to Reproduce

1. Create and init new Wagtail project as usual.
2. Add this minimal code to models.py:

```
class HomePage(Page):
    body = StreamField(block_types=[("doc", DocumentChooserBlock())], blank=True)
    content_panels = [FieldPanel("body")]

class CustomDocument(AbstractDocument):
    my = RichTextField()
    admin_form_fields = ("my",) + Document.admin_form_fields
```
Set custom document model in settings:
`WAGTAILDOCS_DOCUMENT_MODEL = "home.CustomDocument"`

3. Migrate.
4. Login into Wagtail admin.
5. Create new HomePage, add “doc” block.
6. Click “Choose a document”, close dialog by “X”. Do not reload page!
7. Click “Choose a document” again, click “Upload” tab.
8. Observe duplication of field “My”.
9. Repeat steps 6-8 to make more duplication.

In complex project, steps 6-9 are not necessary. RichTextField fields are already duplicated. Maybe some kind of background dialog call?
Looks like it's only for RichTextField.
This is probably true for other dialogs too, such as Image Chooser (I didn't check).

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

Python version: 3.11.9
Django version: 5.0.6
Wagtail version: 6.1.2
Browser version: Firefox 126.0 and Chrome 125.0

### PR DESCRIPTION
**Pull request by laymonage**
Fixes #12002. I recommend toggling whitespace when reviewing.

You can use [my project](https://github.com/laymonage/wagtail-11996-repro) to ease testing.

1. Install with current `main` branch, run migrations
2. Edit a page and add a document chooser block
3. Trigger the chooser a few times
4. Check the Upload tab of the chooser, the rich text field will show up multiple times (and it adds another one every time you open the chooser)
5. Switch to this PR branch
6. Repeat steps 2 and 3
7. The rich text field should only show up once in the Upload tab

### CODE REVIEW

File: client/src/entrypoints/admin/draftail.js [hunk 1]
```diff
@@ -11,52 +11,59 @@ import draftail, {
...
-  }
+    if (!id) {
+      // eslint-disable-next-line no-console
+      console.error('`w-draftail:init` event must have a target with an id.');
+      return;
+    }
 
-  window.draftail.initEditor(`#${id}`, detail, document.currentScript);
-});
+    window.draftail.initEditor(`#${id}`, detail, document.currentScript);
```
Comment (laymonage) [inline client/src/entrypoints/admin/draftail.js:67]: Only the `addEventListener` needs to be run once, but I think we can save some unnecessary computation by wrapping the entire script in the `if` check.

**Review by gasman — APPROVED** [review]
Looks good to me - code change makes sense, and I can confirm it fixes the issue on the test case 👍

**Comment by gasman** [comment]
Merged in a5bb99bf67a5534082a9f7f4f44a33f5086b67f0 (main) and b4ba3d635db90f17eb72903bc743e2f1eb36e4da (stable/6.2.x).

**Comment by lb-** [comment]
Thank you @laymonage

**Pull request closed without merging**